### PR TITLE
Infrastructure

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

**Review by Eabramov84 — APPROVED** [review]

File: include/app.hpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+     *
+     * @return aos::Error
+     */
+    aos::Error Init();
+    /**
+     * Returns Aos application instance.
+     * @return App&
+     */
+
+    static App& Get()
```
Comment (MykolaSolyanko) [inline include/app.hpp:26]: If we use the singleton pattern in the class, I recommend deleting the copy operation

File: include/app.hpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * Aos zephyr application.
+ */
+class App {
+public:
+    /**
+     * Initializes application.
+     *
+     * @return aos::Error
+     */
+    aos::Error Init();
```
Comment (MykolaSolyanko) [inline include/app.hpp:20]: Should we call this function in a constructor? In this case, we won't forget to call this method.

**Review by MykolaSolyanko — APPROVED** [review]

File: include/app.hpp (hunk 2 above)
Comment (al1img) [reply include/app.hpp:20]: We should not. As we don't use the exceptions all necessary initialization should be performed in `Init` function to have possibility to handle errors. See `main` function in this commit.

File: include/app.hpp (hunk 1 above)
Comment (al1img) [reply include/app.hpp:26]: agree

**Pull request merged**